File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max_one = 0;
        int  n = nums.size();
        for(int i = 0; i < n; i++){
            int count1 = 0;
            int x = nums[i];
            if(x == 1){
                while(i < n && nums[i] == 1){
                    count1++;
                    i++;
                }
            } 
            else{
                continue;
            }
            max_one = max(max_one, count1);
        }
        return max_one;
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        n = len(nums)
        ans = 0
        count = 0

        for i in range(n):
            if nums[i] == 1:
                count += 1

            if nums[i] == 0 or i == n-1:
                ans = max(ans, count)
                count = 0
        
        return ans